### ISSUE TITLE
KA10: TOPS-10 BASIC Floating point underflow errors

### ISSUE THREAD

**Issue by pascalgp** [rcornwell/sims#84]
There is a difference in behavior between TOPS-10 6.03 and 7.03 with the following Basic code:

`10 DIM N(3)`
`20 A=0`
`30 N(1)=0`
`40 PRINT A*(0.1-N(1))`
`50 PRINT A*(0.5-N(1))`
`60 END`

Basic 17F under TOPS-10 7.03 on the Simh pdp-10 (KS10) emulator outputs this as expected:

` 0`
` 0`

However, the same version of basic installed from the same tape under TOPS-10 6.03A on the KA10 emulator results in this:

`% UNDERFLOW IN LINE 40`
` 0`
` 2.93874E-39`

**Comment by rcornwell** [comment]
This might be due to some off by one errors I have in my floating point code. These errors occur during random instruction tests.

**Comment by pascalgp** [comment]
Here is another case for reference. This time with FOCAL-10:

TOPS-10 7.03

```
*TYPE FSQT(1000000)
= 1000.0000*
*TYPE FSQT(4.2)
=    2.0494*
*SET A=1.5
*TYPE A
=    1.5000*
```

TOPS-10 6.03

```
*TYPE FSQT(1000000)
= 1027.1907*
*TYPE FSQT(4.2)
%
%Floating-point underflow
TYPE FSQT(4.2)
             ^
=    2.0608*
*SET A=1.5
%
%Floating-point underflow
SET A=1.5
         ^
*TYPE A
=    1.0000*
```

```
.ver
FOCAL 5J(345)-1 +
```

**Comment by rcornwell** [comment]
This should be fixed in e194713. 

For FOCAL please make sure it is built for a KA and not a KL/KS. There are some differences between how the FP works on these processors.

If this works, let me know and I will close this bug.

**Comment by pascalgp** [comment]
Issue is fixed in both BASIC and FOCAL. Thank you. - Closing
